src/authorisation_rules_observer.py

```diff
@@ -74,6 +76,20 @@ def start_authorisation_rules_observer(self):
         # in a hook context, as Juju will disallow use of juju-run.
         new_env = os.environ.copy()
         new_env.pop("JUJU_CONTEXT_ID", None)
+        # Generate the venv path based on the existing lib path
+        for loc in new_env["PYTHONPATH"].split(":"):
+            path = Path(loc)
+            venv_path = (
+                path
+                / ".."
+                / "venv"
+                / "lib"
+                / f"python{version_info.major}.{version_info.minor}"
+                / "site-packages"
+            )
+            if path.stem == "lib":
+                new_env["PYTHONPATH"] = f"{venv_path.resolve()}:{new_env['PYTHONPATH']}"
+                break
```

Comment:
Inject the location of the venv packages when spawning the observer, so that we can use charm deps there (psycopg2). Another option is to try to activate the venv instead.